src/main/java/io/cdap/plugin/servicenow/connector/ServiceNowRecordConverter.java

```diff
@@ -0,0 +1,138 @@
+          try {
+            recordBuilder.setDateTime(fieldName, LocalDateTime.parse(fieldValue, dateTimeFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        case DATE:
+          DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern(DATE_PATTERN);
+          try {
+            recordBuilder.setDate(fieldName, LocalDate.parse(fieldValue, dateFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        case TIME_MILLIS:
+          DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern(TIME_PATTERN);
+          try {
+            recordBuilder.setTime(fieldName, LocalTime.parse(fieldValue, timeFormatter));
+          } catch (DateTimeParseException exception) {
+            throw new UnexpectedFormatException(
+              String.format("Field '%s' of type '%s' with value '%s' is not in ISO-8601 format.",
+                            fieldName, fieldSchema.getDisplayName(), fieldValue), exception);
+          }
+          return;
+        default:
+          throw new UnexpectedFormatException(String.format("Field '%s' is of unsupported type '%s'", fieldName,
+                                                            fieldLogicalType.name().toLowerCase()));
+      }
+    }
+
+    Schema.Type fieldType = fieldSchema.getType();
+    try {
+      switch (fieldType) {
+        case STRING:
+          recordBuilder.set(fieldName, fieldValue);
+          return;
+        case DOUBLE:
+          recordBuilder.set(fieldName, convertToDoubleValue(fieldValue));
+          return;
+        case INT:
+          recordBuilder.set(fieldName, convertToIntegerValue(fieldValue));
+          return;
+        case BOOLEAN:
+          recordBuilder.set(fieldName, convertToBooleanValue(fieldValue));
+          return;
+        default:
+          throw new IllegalStateException(
+            String.format("Record type '%s' is not supported for field '%s'", fieldType.name(), fieldName));
+      }
+    } catch (ParseException exception) {
+      throw new IllegalStateException(
+        String.format("Error occurred while parsing the value '%s' for field '%s'", fieldValue, fieldName));
+    }
+
+  }
+
+  @VisibleForTesting
+  public static Double convertToDoubleValue(String fieldValue) throws ParseException {
+    try {
+      return NumberFormat.getNumberInstance(java.util.Locale.US).parse(fieldValue).doubleValue();
```

Comment:
style: `Locale.US` and import java.util.Locale

Reply:
done cbfb212c77242fe61b2ef14c690a9079b6a9e22f